mod/wikirate_source/spec/set/type/source/source_spec.rb

```diff
@@ -61,182 +85,184 @@
         end
       end
-      context "a non exisiting card link" do
-        it "return errors" do
+      context 'a non exisiting card link' do
+        it 'return errors' do
 
           Card::Env.params[:sourcebox] = 'true'
 
           new_source_url = "#{ Card::Env[:protocol] }#{ Card::Env[:host] }/non_exisiting_card_1"
 
-          new_sourcepage = Card.new :type_id=>Card::SourceID,:subcards=>{ '+Link' => {:content=> new_source_url}, '+File' =>{:type_id=>Card::FileID}, '+Text'=>{:type_id=>Card::BasicID,:content=>""} }
+          new_sourcepage = Card.new source_args(link: new_source_url)
           expect(new_sourcepage).not_to be_valid
           expect(new_sourcepage.errors).to have_key :source
-          expect(new_sourcepage.errors[:source]).to include(" does not exist.")
+          expect(new_sourcepage.errors[:source]).to include(' does not exist.')
 
         end
       end
     end
-    describe "in sourcebox" do
-      context "while link is a card name" do
-        it "returns source card " do
+    describe 'in sourcebox' do
+      context 'while link is a card name' do
+        it 'returns source card ' do
           source_card = create_page
           Card::Env.params[:sourcebox] = 'true'
-          return_source_card = Card.create :type_id=>Card::SourceID,:subcards=>{ '+Link' => {:content=> source_card.name}, '+File' =>{:type_id=>Card::FileID}, '+Text'=>{:type_id=>Card::BasicID,:content=>""} }
+          return_source_card = create_link_source source_card_name
           expect(return_source_card.name).to eq(source_card.name)
         end
-        it "returns error" do
+        it 'returns error' do
           Card::Env.params[:sourcebox] = 'true'
-          return_source_card = Card.new :type_id=>Card::SourceID,:subcards=>{ '+Link' => {:content=> get_a_sample_company.name}, '+File' =>{:type_id=>Card::FileID}, '+Text'=>{:type_id=>Card::BasicID,:content=>""} }
+          return_source_card = Card.new source_args(
+            link: get_a_sample_company.name
+          )
           expect(return_source_card).not_to be_valid
           expect(return_source_card.errors).to have_key :source
-          expect(return_source_card.errors[:source]).to include(" can only be source type or valid URL.")
+          expect(return_source_card.errors[:source])
+            .to include(' can only be source type or valid URL.')
         end
       end
-      context "while link is a non existing card" do
-        it "returns error " do
+      context 'while link is a non existing card' do
+        it 'returns error ' do
           Card::Env.params[:sourcebox] = 'true'
-          return_source_card = Card.new :type_id=>Card::SourceID,:subcards=>{ '+Link' => {:content=> "this is not a exisiting card"}, '+File' =>{:type_id=>Card::FileID}, '+Text'=>{:type_id=>Card::BasicID,:content=>""} }
+          return_source_card = Card.new source_args(
+            link: "this is not a exisiting card"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.